### PR TITLE
griffin: release cursors in cross join

### PR DESCRIPTION
**Pull request by ideoma**
If  `CROSS JOIN` used in PgWire and client disconnects on without receiving full data set then one of the TableReaders is not released. Fix is to ensure that TableReaders are released even when error happens.

### CODE REVIEW

File: core/src/main/java/io/questdb/griffin/engine/join/CrossJoinRecordCursorFactory.java [hunk 1]
```diff
@@ -59,11 +59,18 @@ public void close() {
...
+        RecordCursor masterCursor = masterFactory.getCursor(executionContext);
+        RecordCursor slaveCursor = null;
+        try {
+            slaveCursor = slaveFactory.getCursor(executionContext);
+            if (masterCursor.hasNext()) {
+                cursor.of(masterCursor, slaveCursor);
+                return cursor;
+            }
+        } catch (Throwable ex) {
+            Misc.free(masterCursor);
```
Comment (puzpuzpuz) [inline core/src/main/java/io/questdb/griffin/engine/join/CrossJoinRecordCursorFactory.java:71]: I think we need a similar catch block in `getCursor` method in the following cursor factories:
* `AsOfJoinLightRecordCursorFactory`
* `AsOfJoinNoKeyRecordCursorFactory`
* `AsOfJoinRecordCursorFactory`
* `HashJoinLightRecordCursorFactory` (this one has a catch block, but the try block doesn't include the master factory's `getCursor` call)
* `HashJoinRecordCursorFactory` (same as above)
* `HashOuterJoinLightRecordCursorFactory` (same as above)
* `HashOuterJoinRecordCursorFactory` (same as above)
* `LtJoinLightRecordCursorFactory`
* `LtJoinNoKeyRecordCursorFactory`
* `LtJoinRecordCursorFactory`
* `SpliceJoinLightRecordCursorFactory`
* `UnionAllRecordCursorFactory`

Comment (ideoma) [reply core/src/main/java/io/questdb/griffin/engine/join/CrossJoinRecordCursorFactory.java:71]: Good spot, fixed those too

**Review by puzpuzpuz — APPROVED** [review]

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 100 / 108 (92.59%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/engine/join/AsOfJoinRecordCursorFactory.java|5|9|55.56%|
|:large_blue_circle:|io/questdb/griffin/engine/join/HashJoinRecordCursorFactory.java|4|6|66.67%|
|:large_blue_circle:|io/questdb/griffin/engine/join/HashOuterJoinRecordCursorFactory.java|4|6|66.67%|
|:large_blue_circle:|io/questdb/griffin/engine/join/LtJoinNoKeyRecordCursorFactory.java|9|9|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/join/AsOfJoinNoKeyRecordCursorFactory.java|9|9|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/union/UnionAllRecordCursorFactory.java|9|9|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/join/LtJoinRecordCursorFactory.java|9|9|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/join/SpliceJoinLightRecordCursorFactory.java|9|9|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/join/HashOuterJoinLightRecordCursorFactory.java|6|6|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/join/HashJoinLightRecordCursorFactory.java|6|6|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/join/CrossJoinRecordCursorFactory.java|10|10|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/join/LtJoinLightRecordCursorFactory.java|9|9|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/join/AsOfJoinLightRecordCursorFactory.java|11|11|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 100 / 108 (92.59%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/engine/join/AsOfJoinRecordCursorFactory.java|5|9|55.56%|
|:large_blue_circle:|io/questdb/griffin/engine/join/HashJoinRecordCursorFactory.java|4|6|66.67%|
|:large_blue_circle:|io/questdb/griffin/engine/join/HashOuterJoinRecordCursorFactory.java|4|6|66.67%|
|:large_blue_circle:|io/questdb/griffin/engine/join/LtJoinNoKeyRecordCursorFactory.java|9|9|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/join/AsOfJoinNoKeyRecordCursorFactory.java|9|9|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/union/UnionAllRecordCursorFactory.java|9|9|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/join/LtJoinRecordCursorFactory.java|9|9|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/join/SpliceJoinLightRecordCursorFactory.java|9|9|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/join/HashOuterJoinLightRecordCursorFactory.java|6|6|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/join/HashJoinLightRecordCursorFactory.java|6|6|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/join/CrossJoinRecordCursorFactory.java|10|10|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/join/LtJoinLightRecordCursorFactory.java|9|9|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/join/AsOfJoinLightRecordCursorFactory.java|11|11|100.00%|

**Pull request merged**